lib/shared/drive-constraints.js

```diff
@@ -208,3 +208,115 @@ exports.isDriveValid = (drive, image) => {
 exports.isDriveSizeRecommended = (drive, image) => {
   return _.get(drive, [ 'size' ], UNKNOWN_SIZE) >= _.get(image, [ 'recommendedDriveSize' ], UNKNOWN_SIZE);
 };
+
+/**
+ * @summary Drive/image compatibility status messages.
+ * @public
+ *
+ * @description
+ * Status messages intended to be shown to the user.
+ *
+ * @type {Object}
+ */
+exports.COMPATIBILITY_STATUS_MESSAGES = {
+  OK: '',
+  NOT_RECOMMENDED: 'Not Recommended',
+  TOO_SMALL: 'Too Small For Image',
+  LOCKED: 'Locked',
+  SYSTEM: 'System Drive',
+  CONTAINS_IMAGE: 'Drive Contains Image'
+};
+
+/**
+ * @summary Drive/image compatibility status types.
+ * @public
+ *
+ * @description
+ * Status types classifying what kind of message it is, i.e. ok, error, warning.
+ *
+ * @type {Object}
+ */
+exports.COMPATIBILITY_STATUS_TYPES = {
+  OK: 0,
+  ERROR: 1,
+  WARNING: 2
+};
+
+/**
+ * @summary Get drive feedback in an object, if any
+ * @function
+ * @public
+ *
+ * @description
+ * If the drive-image combination results in a warning or error, this function
+ * returns an object containing the feedback type ('error', 'warning') and the
+ * feedback message.
+ *
+ * @param {Object} drive - drive
+ * @param {Object} image - image
+ * @returns {Object} feedback object
+ *
+ * @example
+ * const drive = {
+ *   device: '/dev/disk2',
+ *   name: 'My Drive',
+ *   size: 4000000000
+ * };
+ *
+ * const image = {
+ *   path: 'rpi.img',
+ *   size: 2000000000
+ *   recommendedDriveSize: 4000000000
+ * });
+ *
+ * const { type, message } = constraints.getDriveImageCompatibilityStatus(drive, image);
+ *
+ * if (type === constraints.COMPATIBILITY_STATUS_TYPES.WARNING) {
+ *   // do something
+ * } else if (type === constraints.COMPATIBILITY_STATUS_TYPES.ERROR) {
+ *   // do something else
+ * }
+ */
+exports.getDriveImageCompatibilityStatus = (drive, image) => {
+
+  // Mind the order of the if-statements if you modify.
+  if (exports.isSourceDrive(drive, image)) {
+    return {
```

Comment:
I think its fine as it is. Storing the object instances somewhere looks like to much hassle to me.